### ISSUE TITLE
Support for marked-up text when audio is not available?

### ISSUE THREAD

**Issue by stripethree** [mandnyc/ssml-builder#7]
I am working on an Action for Google and upon searching for some kind of SSML builder in JS, decided to give this project a try. Finding it very useful so far, 👍 for that.

The SSML spec for [audio](https://www.w3.org/TR/speech-synthesis/#S3.3.1) allows for content/children to be spoken in the case that the `src` is not available. In the case of the Action I am developing, this would be very useful.

Before I go off and write something and maybe even create a PR, I wondered what consideration has already been given to support this capability. Also curious on thoughts of how comprehensive to make such support since "the alternate content may include text, speech markup, `desc` elements, or other `audio` elements." That can get pretty complicated.

**Comment by mandnyc** [comment]
That’s great to know. When I built this, it was for Amazon originally and I didn’t know this could be done since I didn’t read the spec for SSML fully. I have an idea about this and I’ll share some thoughts soon on it. Whats your thoughts? @stripethree

**Comment by stripethree** [comment]
The simplest solution in my mind would start with an additional, optional parameter that is a `string`. If provided, it becomes the content of the tag:

```
const SSMLBuilder = require('ssml-builder');
const builder = new SSMLBuilder();

builder.audio('http://domain.com/audio.mp3', 'Sorry, the audio is not available');
```
Result:
```
  <audio src="http://domain.com/audio.mp3">  
    Sorry, the audio is not available.
  </audio>
```

This doesn't really give the full spec though. So I thought about having that second argument be another instance of `ssml-builder`:

```
const SSMLBuilder = require('ssml-builder');
const builder = new SSMLBuilder();
const audioBuilder = new SSMLBuilder();

audioBuilder.say('Sorry.');
audioBuilder.pause('1s');
audioBuilder.say('The audio is not available.');

builder.audio('http://domain.com/audio.mp3', audioBuilder);
```

Under the hood, the `audio` function would call `ssml` with a `false` argument, so that the `speak` tags are not rendered, so the result would look like this:

```
  <audio src="http://domain.com/audio.mp3">  
    Sorry. <break time='1s'/> The audio is not available.
  </audo>
```

The upside of this approach is the spec is more fully supported. Application devs could reuse the `audioBuilder` instance if they have a common error message that they want to use for multiple audio tags. The downside is having to manage multiple `ssml-builder` instances in the application and potentially introducing super complex speech tags that might actually be supported; testing that to exhaustion would be a challenge.

I think if I had to pick between the two, I lean towards the simple solution since it is a starting point that will not introduce a lot of complexity or potential breakage. Still mulling alternatives though and am definitely open to more/other ideas.

**Comment by koushinrin** [comment]
Waiting for this feature

**Comment by mandnyc** [comment]
@stripethree Hi Jeff, I'm very sorry for my delay. I've been occupied by work. I gave some thoughts to your two suggestions. I do like your second approach but I think we can make it easier for the users. Instead of having the users to create another instance of the Speech object, I think the second parameter could be a callback function that takes an instance of the Speech object that we can create for the users. Here is how I think it would look,

```javascript
builder.audio('http://domain.com/audio.mp3', function(audioBuilder) {
        audioBuilder.say('Sorry.');
        audioBuilder.pause('1s');
        audioBuilder.say('The audio is not available.');
});
```
Going back to your point about not knowing how Amazon and Google would support it, we could just fix the bugs as they come along.

**Comment by mandnyc** [comment]
@koushinrin @stripethree What do you think about this approach?

**Comment by rburton** [comment]
I saw your Tweet (https://twitter.com/MandyChanNYC/status/913241419569037312) @mandnyc and hunted your profile down on Github. 

I opened a Pull Request (#8)  based on your suggestion.

It was very straightforward, especially after your exchange with @stripethree. The one thing I did stumble on a little was `Speech#ssml(boolean)`. This parameter is a boolean for whether or not the `<speak/>` should be excluded. Initially, I passed in false thinking it would exclude the speak tag. The opposite is actually true; passing in true excludes the speak tag.

I also added node_modules to your .gitignore. Let me know if you're okay with that. Most projects exclude the node_modules folder and require the developer to run `npm install` to download the dependencies.

I ran `npm test` and all the tests pasted!

Let me know if the PR is okay or if I should make any changes.
